src/setup_node_env/heap_snapshot.js

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+var getopts = require('getopts');
+var v8 = require('node:v8');
+var worker = require('node:worker_threads');
+
+var execOpts = getopts(process.execArgv);
+var envOpts = getopts(process.env.NODE_OPTIONS ? process.env.NODE_OPTIONS.split(/\s+/) : []);
+var diagnosticDir = execOpts['diagnostic-dir'] || envOpts['diagnostic-dir'];
+var heapSnapshotSignal = execOpts['heapsnapshot-signal'] || envOpts['heapsnapshot-signal'];
+var heapSnapshotSerial = 0;
+
+function getHeapSnapshotPath() {
+  var now = new Date();
+
+  var year = now.getFullYear();
+  var month = String(now.getMonth() + 1).padStart(2, '0');
+  var day = String(now.getDate()).padStart(2, '0');
+  var hours = String(now.getHours()).padStart(2, '0');
+  var minutes = String(now.getMinutes()).padStart(2, '0');
+  var seconds = String(now.getSeconds()).padStart(2, '0');
+
+  var date = `${year}${month}${day}`;
+  var time = `${hours}${minutes}${seconds}`;
+  var pid = process.pid;
+  var threadId = worker.threadId;
+  var serial = (++heapSnapshotSerial).toString().padStart(3, '0');
+
+  return `${diagnosticDir}/Heap.${date}.${time}.${pid}.${threadId}.${serial}.heapsnapshot`;
```

Comment:
Maybe path.join instead of `/`? I'm not sure if writeHeapSnapshot is going to fix Windows paths and don't have a machine to quickly test on.

Reply:
Just slipped my mind, makes total sense 👍